src/main/java/com/google/sps/utilities/RealTutorSessionDatastore.java

```diff
@@ -0,0 +1,171 @@
+import com.google.appengine.api.datastore.Query.CompositeFilter;
+import java.lang.String;
+import java.util.List;
+import java.util.ArrayList;
+import java.util.Calendar;
+import com.google.gson.Gson;
+
+/** Accesses Datastore to manage tutoring sessions. */ 
+public final class RealTutorSessionDatastore implements TutorSessionDatastoreService {
+
+    /**
+    * Adds a new TutorSession for the tutor and student.
+    */
+    @Override
+    public void addTutorSession(String tutorEmail, String studentEmail, TutorSession session) {
+        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+        TransactionOptions options = TransactionOptions.Builder.withXG(true);
+        Transaction txn = datastore.beginTransaction(options);
+
+        try {
+
+            Entity sessionEntity = new Entity("TutorSession");
+
+            sessionEntity.setProperty("tutorEmail", tutorEmail.toLowerCase());
+            sessionEntity.setProperty("studentEmail", studentEmail.toLowerCase());
+            sessionEntity.setProperty("subtopics", session.getSubtopics());
+            sessionEntity.setProperty("questions", session.getQuestions());
+            sessionEntity.setProperty("isRated", session.isRated());
+            sessionEntity.setProperty("rating", session.getRating());
+            sessionEntity.setProperty("timeslot", updateTimeRange(tutorEmail, session.getTimeslot(), datastore, txn));
+
+            datastore.put(txn, sessionEntity);
+
+            txn.commit();
+
+        } finally {
+          if (txn.isActive()) {
+            txn.rollback();
+          }
+        }
+    }
+
+    /**
+    * Gets a list of all scheduled sessions for a tutor with the given email.
+    * @return List<TutorSession>
+    */
+    @Override
+    public List<TutorSession> getScheduledSessionsForTutor(String email) {
+        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+        return getScheduledSessions(datastore, "tutor", email);
+    }
+
+    /**
+    * Gets a list of all scheduled sessions for a student with the given email.
+    * @return List<TutorSession>, empty list if the tutor does not exist
+    */
+    @Override
+    public List<TutorSession> getScheduledSessionsForStudent(String email) {
+        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+        return getScheduledSessions(datastore, "student", email);
+    }
+
+    /**
+    * Gets all the tutor session entities for a userType (tutor or student) with the corresponding email. 
+    * @return ArrayList<TutorSession>, empty list if the student does not exist
+    */
+    private ArrayList<TutorSession> getScheduledSessions(DatastoreService datastore, String userType, String email) {
+        ArrayList<TutorSession> sessions = new ArrayList<TutorSession>();
+
+        //get all sessions with given email
+        Filter filter = new FilterPredicate(userType + "Email", FilterOperator.EQUAL, email.toLowerCase());
```

Comment:
How does this make the distinction between the sessions a user is a tutor and the ones a user is a student?

Reply:
The `userType` variable is going to either be "student" or "tutor". So if `userType` is "student", it will filter by "studentEmail" and if the `userType` is "tutor", it will filter by "tutorEmail".